craft_application/services/lifecycle.py

```diff
@@ -135,6 +143,7 @@ def __init__(  # noqa: PLR0913 (too many arguments)
     def setup(self) -> None:
         """Initialize the LifecycleManager with previously-set arguments."""
         self._lcm = self._init_lifecycle_manager()
+        callbacks.register_post_step(self.post_prime, step_list=[Step.PRIME])
```

Comment:
there's a (possibly minor) issue here with this reference to `self.post_prime` keeping `self` alive forever (which in turn keeps the lifecycle manager alive forever). It's not necessarily a concern on the applications because typically the process finishes soon after the single lifecycle run is executed, but I thought I'd point it out

I don't have any good suggestions for this though